src/controllers/book.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from 'express';
+import { SortOrder } from 'mongoose';
+import { Book } from '../models/book';
+import { IBookQuery } from '../interfaces/book-query';
+
+import { bookService } from '../services/book-service';
+import { bookValidation } from '../validation/book-validation';
+import { authValidation } from '../validation/auth-validation';
+
+class BookController {
+  public async create(req: Request, res: Response): Promise<void> {
+    authValidation.ensureLoggedIn(req.userRole, req.user);
+    const book = await new Book({
+      ...req.body,
+      user_id: req.user._id
+    }).save();
+
+    res.status(201).send({
+      success: true,
+      data: { message: 'New book created successfully', book }
+    });
+  }
+
+  public async update(req: Request, res: Response): Promise<void> {
+    authValidation.ensureLoggedIn(req.userRole, req.user);
+    bookValidation.ensureValidEditFields(req, res);
+
+    const book = await bookService.update(
+      req.params.id,
+      req.user._id,
+      req.body
+    );
+
+    if (!book) {
+      bookValidation.sendBookNotFound(req, res)
+    } else {
+      bookValidation.sendSuccess(res, book)
+    }
```

Comment:
Invert if to make it simpler
```suggestion
    if (book) {
     sendSuccess(res, book)
    } else {
     sendBookNotFound(req, res)
    }
```

And please put the two methods here, in this class. They are not "validation"

Reply:
Don't give the method sendBookNotFound the object req